### PR TITLE
feat: (sfui2-1039) category list

### PR DESCRIPTION
**Pull request by aniamusial**
# Related issue

https://vsf.atlassian.net/browse/SFUI2-1039?atlOrigin=eyJpIjoiOTc2NDI2M2Y3ZTJkNGFjZjk5N2Q0MzMxOTlmNGViZDYiLCJwIjoiaiJ9

# Scope of work

- created category list block
- updated filters panel block

# Screenshots of visual changes

Zrzut ekranu 2023-05-10 o 10 09 14
Zrzut ekranu 2023-05-10 o 10 09 04

# Checklist

- [ ] Self code-reviewed
- [x] Changes documented
- [x] Semantic HTML
- [x] SSR-friendly
- [x] Caching friendly
- [x] a11y for WCAG 2.0 AA
- [x] examples created
- [x] blocks created
- [x] cypress tests created

### CODE REVIEW

File: apps/preview/next/pages/showcases/Filters/CategoryList.tsx [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+              as="a"
+              href={category.link}
+              className={classNames('first-of-type:mt-2 rounded-md active:bg-primary-100', {
+                'bg-primary-100 hover:bg-primary-100': isCategorySelected(category.label),
+              })}
+              slotSuffix={isCategorySelected(category.label) && <SfIconCheck size="xs" className="text-primary-700" />}
+              onClick={() => handleCategorySelection(category.label)}
+            >
+              <span className="flex items-center">
+                <span className="text-left">{category.label}</span>
```
Comment (Szymon-dziewonski) [inline apps/preview/next/pages/showcases/Filters/CategoryList.tsx:91]:
```suggestion
              <span className="flex items-center">
                {category.label}
```
No need to wrap it, same for react

File: apps/preview/next/pages/showcases/Filters/FiltersPanel.tsx [hunk 2]
```diff
@@ -216,6 +263,37 @@ export default function FiltersPanel() {
                 ))}
               </ul>
             )}
+            {section.type === 'categories' && (
+              <ul className="mt-2 mb-6">
+                <li>
+                  <SfListItem size="sm" as="button" type="button">
+                    <div className="flex items-center">
+                      <SfIconArrowBack size="sm" className="text-neutral-500" />
+                      <p className="ml-3">Back to {section.details[0].label}</p>
```
Comment (Szymon-dziewonski) [inline apps/preview/next/pages/showcases/Filters/FiltersPanel.tsx:272]:
```suggestion
                      <SfIconArrowBack size="sm" className="text-neutral-500 mr-3" />
                      Back to {section.details[0].label}
```

File: apps/preview/nuxt/pages/showcases/Filters/FiltersPanel.vue [hunk 3]
```diff
@@ -235,6 +326,13 @@ const handleSingleSelection = (val: string) => {
   newSelectedFilters.push(val);
   selectedFilters.value = newSelectedFilters;
 };
+const handleCategorySelection = (val: string) => {
```
Comment (Szymon-dziewonski) [inline apps/preview/nuxt/pages/showcases/Filters/FiltersPanel.vue:329]:
```suggestion
const handleCategorySelection = (selectedValue: string) => {
```

File: apps/preview/nuxt/pages/showcases/Filters/CategoryList.vue [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+    counter: 52,
+    link: '#',
+  },
+]);
+
+const open = ref(true);
+const selectedCategories = ref<string[]>([]);
+
+const isCategorySelected = (val: string) => selectedCategories.value.includes(val);
+const handleCategorySelection = (val: string) => {
```
Comment (Szymon-dziewonski) [inline apps/preview/nuxt/pages/showcases/Filters/CategoryList.vue:90]:
```suggestion
const handleCategorySelection = (selectedValue: string) => {
```
IMHO suggestion is more descriptive

**Comment by aniamusial** [comment]
> I'm not sure if active/hover on `back to clothing` should be rounded https://github.com/vuestorefront/storefront-ui/blob/37322a895eb58edb21c313dfd48d7dd7e606353f/apps/docs/components/.vuepress/components/Showcase.vue probably something to ask Jacek
> 
> Should clear filters always be visible, IMHO it would be nice if that would hidden, but I'm not sure :D

Yes, it should always be visible, according to Jacek and changes we agreed on some time ago - I just removed the second, additional "Clear all" as it was invalid :)

**Comment by Szymon-dziewonski** [comment]
@aniamusial Cool so lets keep it as Jacek wanted 

What about 

> I'm not sure if active/hover on back to clothing should be rounded https://github.com/vuestorefront/storefront-ui/blob/37322a895eb58edb21c313dfd48d7dd7e606353f/apps/docs/components/.vuepress/components/Showcase.vue probably something to ask Jacek

**Comment by aniamusial** [comment]
@Szymon-dziewonski according to Jacek it should stay the way it is now

**Review by Szymon-dziewonski — APPROVED** [review]

**Pull request merged**